### PR TITLE
Only consider *.sql files when applying schema.

### PR DESCRIPTION
**Pull request by dominichamon**
Also add more checks and clearer error messages.

h/t to @michael-berlin

### CODE REVIEW

**Comment by dominichamon** [comment]
@alainjobart @michael-berlin

File: py/vttest/local_database.py [hunk 1]
```diff
@@ -94,20 +95,25 @@ def load_schema(self):
...
-              'Loading schema for keyspace %s from file %s', keyspace,
-              filepath)
-          cmds = self.get_sql_commands_from_file(filepath, keyspace_dir)
-
-          # Run the cmds on each shard in the keyspace.
-          for shard in self.shards:
-            if shard.keyspace == keyspace:
-              self.mysql_execute(cmds, db_name=shard.db_name)
+      if not os.path.isdir(keyspace_dir):
+        raise Exception('No keyspace subdirectory found in the schema dir. '
```
Comment (enisoc) [inline py/vttest/local_database.py]: There might be a test case where you'd create a keyspace without any initial schema. Can we make this log a warning and continue to the next keyspace?

Comment (dominichamon) [inline py/vttest/local_database.py]: is it valid to set a schema_dir in that case?

**Comment by alainjobart** [comment]
This LGTM, but I'm unclear on relationship with:
https://github.com/youtube/vitess/pull/1233
? does this contain both?

File: py/vttest/local_database.py (hunk 1 above)
Comment (enisoc) [inline py/vttest/local_database.py]: Your test could use multiple keyspaces, only some of which have initial schema.

Comment (michael-berlin) [inline py/vttest/local_database.py]: After discussing it offline with Anthony:

Let's keep the hard failure here.

Instead, users can create empty directories to work-around this problem.

Can you please mention this in the exception e.g.:

``` py
raise Exception('No keyspace subdirectory found in the schema dir.'
                ' Expected directory: %s' % keyspace_dir)
                ' For keyspaces without an initial schema, create the'
                ' directory and leave a README file to explan why the'
                ' directory exists.')
```

**Comment by michael-berlin** [comment]
Thanks for the pull request!

They'll show up internally with the next import.

**Pull request merged**